js/viewer/viewer.js

```diff
@@ -908,7 +912,7 @@ function Viewer(parent, config) {
                 var tracker = this.mouseTrackers.shift();
                 var userData = tracker.userData;
 
-                if(userData){
+                if(userData && userData.type != 'POLYGON'){
                     this.dispatchSVGEvent("removeAnnotationByGraphID", {
```

Comment:
This is not needed, your first `if` is supposed to handle polygon case.

Reply:
I also thought so, but this `userData.type != 'POLYGON'` is needed so that when I clicked on any other tool after the polygon drawing is finished (`drawingStop` is called from annotation-tool.js), the polygon isn't deleted.

The flow for when a different button is clicked is:
1. drawingStop - this function calls `removeMouseTrackers` which removes all the temp SVG objects that are created (empty objects), but since we are not adding this empty object in case of `polygon` we need to make sure that the last polygon object (valid one) is not deleted.
2. update drawing mode

Reply:
Right, my bad. I somehow thought this part of the `onMouseDragToDrawEnd` function